frontend/src/resources/role-assignment.ts

```diff
@@ -0,0 +1,35 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { Metadata } from './metadata'
+import { IResource, IResourceDefinition } from './resource'
+import { Subject } from './kubernetes-client'
+
+export const RoleAssignmentApiVersion = 'rbac.open-cluster-management.io/v1alpha1'
+export type RoleAssignmentApiVersionType = 'rbac.open-cluster-management.io/v1alpha1'
+
+export const RoleAssignmentKind = 'RoleAssignment'
+export type RoleAssignmentKindType = 'RoleAssignment'
+
+export const RoleAssignmentDefinition: IResourceDefinition = {
+  apiVersion: RoleAssignmentApiVersion,
+  kind: RoleAssignmentKind,
+}
+
+export interface Cluster {
+  name: string
+  clusterWide: boolean
+  namespaces?: string[]
+}
+
+export interface RASubject extends Subject {
```

Comment:
`RoleAssignmentSubject` is a bit more descriptive FMPOV
```suggestion
export interface RoleAssignmentSubject extends Subject {
```